src/tlo/methods/contraception.py

```diff
@@ -644,27 +646,103 @@ def select_contraceptive_following_birth(self, mother_id, mother_age):
+             get_item_code("IUD, Copper T-380A"): 1}
+        _cons_codes['injections'] = \
+            {get_item_code("Depot-Medroxyprogesterone Acetate 150 mg - 3 monthly"): 1,
+             get_item_code("Glove disposable powdered latex medium_100_CMST"): 1,
+             get_item_code("Water for injection, 10ml_Each_CMST"): 1,
+             get_item_code("Povidone iodine, solution, 10 %, 5 ml per injection"): 5,
+             get_item_code("Gauze, swabs 8-ply 10cm x 10cm_100_CMST"): 1}
+        _cons_codes['implant'] =\
+            {get_item_code("Glove disposable powdered latex medium_100_CMST"): 3,
+             get_item_code("Lidocaine HCl (in dextrose 7.5%), ampoule 2 ml"): 2,
+             get_item_code("Povidone iodine, solution, 10 %, 5 ml per injection"): 1*5,  # unit: 1 ml
+             get_item_code("Syringe, needle + swab"): 2,
+             get_item_code("Trocar"): 1,
+             get_item_code("Needle suture intestinal round bodied ½ circle trocar_6_CMST"): 1,
+             get_item_code("Jadelle (implant), box of 2_CMST"): 1,  # implant used in 50% cases
+             get_item_code("Implanon (Etonogestrel 68 mg)"): 1,  # alternative implant used in other 50% cases
+             # (see _cons_alternatives_probs)
+             get_item_code("Gauze, swabs 8-ply 10cm x 10cm_100_CMST"): 1}
+        # Implants are alternatives, type chosen according to probabilities when administrated.
+        _cons_alternatives_probs['implant'] = \
+            {get_item_code("Jadelle (implant), box of 2_CMST"): 0.5,
+             get_item_code("Implanon (Etonogestrel 68 mg)"): 0.5}
+        _cons_codes['female_sterilization'] =\
+            {get_item_code("Lidocaine HCl (in dextrose 7.5%), ampoule 2 ml"): 1,
+             get_item_code("Atropine sulphate  600 micrograms/ml, 1ml_each_CMST"): 1,  # used only in 50% cases
+             # (see _cons_alternatives_probs)
+             get_item_code("Diazepam, injection, 5 mg/ml, in 2 ml ampoule"): 1,
+             get_item_code("Syringe, autodestruct, 5ml, disposable, hypoluer with 21g needle_each_CMST"): 3,
+             get_item_code("Gauze, swabs 8-ply 10cm x 10cm_100_CMST"): 2,
+             get_item_code("Needle, suture, assorted sizes, round body"): 3,
+             get_item_code("Suture, catgut, chromic, 0, 150 cm"): 3,
+             get_item_code("Tape, adhesive, 2.5 cm wide, zinc oxide, 5 m roll"): 125,  # unit: 1 cm long (2.5 cm wide)
+             get_item_code("Glove surgeon's size 7 sterile_2_CMST"): 2,
+             get_item_code("Paracetamol, tablet, 500 mg"): 8*500,  # unit: 1 mg
+             get_item_code("Povidone iodine, solution, 10 %, 5 ml per injection"): 2*5,  # unit: 1 ml
+             get_item_code("Cotton wool, 500g_1_CMST"): 100}  # unit: 1 g
+        # Atropine is used only in some cases
+        _cons_alternatives_probs['female_sterilization'] = \
+            {-99: 0.5,  # no alternative used
```

Comment:
for female_sterilization and pills and implants ----- check what the EFFECTIVE probability of (essential) consumables being available is in `master` versus this branch.